octopus-agent/src/main/java/octopus/teamcity/agent/buildinformation/OctopusBuildInformationBuildProcess.java

```diff
@@ -52,9 +52,16 @@ public OctopusBuildInformationBuildProcess(
     final Map<String, String> sharedConfigParameters = runningBuild.getSharedConfigParameters();
 
     final BuildInfoUserData buildInfoUserData = new BuildInfoUserData(parameters);
-    final String buildId = Long.toString(runningBuild.getBuildId());
 
-    final URL buildUrl = constructBuildUrl(runningBuild, buildId);
+    String buildUrlString = sharedConfigParameters.get("externalBuildUrl");
+    if (buildUrlString == null) {
```

Comment:
nit: I'd put all this into the constructBuildUrl - and hide that there's logic around how its contstruted